rules/TypeDeclaration/TypeInferer/SilentVoidResolver.php

```diff
@@ -44,15 +46,14 @@ public function hasExclusiveVoid(ClassMethod | Closure | Function_ $functionLike
             return false;
         }
 
-        /** @var Return_[] $returns */
-        $returns = $this->betterNodeFinder->findInstancesOfInFunctionLikeScoped($functionLike, Return_::class);
-        foreach ($returns as $return) {
-            if ($return->expr instanceof Expr) {
-                return false;
+        $return = $this->betterNodeFinder->findFirstInFunctionLikeScoped($functionLike, function(Node $node) {
+            if ($node instanceof Return_ && $node->expr instanceof Expr) {
+                return true;
             }
-        }
+            return false;
```

Comment:
`findFirstInFunctionLikeScoped` can return `false` or `true` depends of which first found, it is better to use :

```php
$found = false;
$this->simpleCallableNodeTraverser->traverseNodesWithCallable(
    (array) $functionLike->stms,
    function ($subNode) use (&$found) {
            if ($subNode instanceof Class_ || $subNode instanceof Function_ || $subNode instanceof Closure) {
                  return NodeTraverser::DONT_TRAVERSE_CURRENT_AND_CHILDREN;
            }
            
           if ($subNode instanceof Return_ && $subNode->expr instanceof Expr) {
                $found = true;
                return NodeTraverser::STOP_TRAVERSAL; 
           }
           
           return null;
   }
);

return $found;
```